src/plugins/data_source/server/legacy/http_aws_es/connector.js

```diff
@@ -98,6 +98,9 @@ class HttpAmazonESConnector extends HttpConnector {
       hostname: this.endpoint.hostname,
     });
 
+    // copy across params
+    Object.assign(request, reqParams);
+
```

Comment:
will reqParams contains `endpoint` or `hostname`?

Reply:
Not likely, reqParams is supposed to contain only query / body / path, and actually it is a existing logic in https://github.com/opensearch-project/OpenSearch-Dashboards/pull/9641/files#diff-1a4e548980eeca5855fdea8b28f692f1becf52b524a6bfe40573c0d6aed29ebcL111-L112 . The previous PR forget to add it back while upgrading aws-sdk.